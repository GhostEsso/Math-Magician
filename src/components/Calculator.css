* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

body {
  height: auto;
}

.Calc-container {
  display: flex;
  flex-direction: column;
  width: 40%;
  height: 45vh;
  margin: 20vh 20vw;
  box-shadow: 0 0 10px rgba(143, 133, 133, 0.726);
}

.row {
  display: flex;
  flex-direction: row;
  align-items: stretch;
  height: 100%;
}

.row input[type="button"] {
  flex: 1;
  border: 0.5px solid rgb(193, 207, 216);
  color: #050000;
  font-weight: 600;
}

.input::-webkit-outer-spin-button,
.input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.Calc-container .input {
  background-color: rgb(216, 88, 88);
  width: 100%;
  border: none;
  outline: none;
  font-size: 27px;
  text-indent: 5px;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: right;

  /* text-align: right; */
  padding-right: 5px;
}

.last-row {
  display: flex;
  flex-direction: row;
}

.last-row input[type="button"]:first-child {
  flex: 2;
}

.row .input:first-child {
  background-color: rgb(122, 129, 127);
}

.row input:last-child {
  background-color: rgb(233, 174, 13);
}

.Quotes {
  display: flex;
  margin-top: 15%;
  padding: 1% 20%;
  font-family: 'Courier New', Courier, monospace;
  font-size: large;
  color: rgb(0, 0, 0);
}

.Calculator {
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  font-size: larger;
  padding: 50px;
}
