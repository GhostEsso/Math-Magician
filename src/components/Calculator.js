import './Calculator.css';
import CalApp from './CallApp';

const Calculator = () => (
  <CalApp />
);
Calculator.defaultProps = {
  num: 0,
};
export default Calculator;