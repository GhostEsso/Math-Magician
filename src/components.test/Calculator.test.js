import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Snapshot test for Calculator component', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
