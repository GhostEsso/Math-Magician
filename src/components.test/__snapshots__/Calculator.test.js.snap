// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Calculator component 1`] = `
<section
  className="Calculator"
>
  <p>
    Let's do some math!
  </p>
  <section
    className="Calc-container"
  >
    <div
      className="row"
    >
      <h2
        className="input"
      >
        0
      </h2>
    </div>
    <div
      className="row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        onClick={[Function]}
        type="button"
        value="รท"
      />
    </div>
    <div
      className="row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        onClick={[Function]}
        type="button"
        value="x"
      />
    </div>
    <div
      className="row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        onClick={[Function]}
        type="button"
        value="-"
      />
    </div>
    <div
      className="row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="row last-row"
    >
      <input
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </section>
</section>
`;
